run/events-pubsub/Startup.cs

```diff
@@ -46,38 +45,47 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<
         {
             endpoints.MapPost("/", async context =>
             {
-                logger.LogInformation("Handling HTTP POST");
-
-                using (var reader = new StreamReader(context.Request.Body))
+                var cloudEvent = await context.Request.ReadCloudEventAsync();
+                if (cloudEvent == null)
                 {
-                    var body = await reader.ReadToEndAsync();
-                    logger.LogInformation($"HTTP POST body: {body}");
+                    context.Response.StatusCode = 400;
+                    await context.Response.WriteAsync("Bad request: No CloudEvent received");
+                    return;
+                }
+                logger.LogInformation("Received CloudEvent\n" + GetEventLog(cloudEvent));
 
-                    dynamic cloudEventData = JsonConvert.DeserializeObject(body);
-                    if (cloudEventData == null)
-                    {
-                        context.Response.StatusCode = 400;
-                        await context.Response.WriteAsync("Bad request: No Pub/Sub message received");
-                        return;
-                    }
+                var messagePublishedData = CloudEventConverters.ConvertCloudEventData<MessagePublishedData>(cloudEvent);
+                var pubSubMessage = messagePublishedData?.Message;
+                if (messagePublishedData == null || pubSubMessage == null)

```

Comment:
nit: no need to ask for messagePublishData here.
```suggestion
                if (pubSubMessage == null)

```

Reply:
Good point, I'll fix

Reply:
Turns out, `messagePublishedData` is never null. It can just be empty if it's not the right format. So, I changed the code to only check for `pubSubMessage` null, which is possible if there is not `message` field. Also, `cloudEvent` cannot be null either, even if the HTTP Post is not in CloudEvent format. So, I removed the null check there as well.